web/modules/custom/joinup_invite/src/Form/InviteFormBase.php

```diff
@@ -46,145 +49,237 @@ public static function create(ContainerInterface $container) {
-    parent::validateForm($form, $form_state);
-    $triggering_element = $form_state->getTriggeringElement();
-    $users = empty($form_state->getValue('users')) ? [] : array_filter($form_state->getValue('users'));
-    if ($triggering_element['#name'] !== 'op_filter' && empty($users)) {
-      $form_state->setErrorByName('users', $this->t('You must select at least one user'));
+    if (empty($form_state->get('user_list'))) {
+      $form_state->setError($form, $this->t('Please add at least one user.'));
     }
   }
 
+  /**
+   * Ajax callback that returns the updated users list.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   A render array.
+   */
+  public function ajaxUpdateUserList(array $form, FormStateInterface $form_state) {
+    return $form['users'];
+  }
+
+  /**
+   * Ajax callback that returns the updated form.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   A render array.
+   */
+  public function ajaxUpdateForm(array $form, FormStateInterface $form_state) {
+    return $form;
+  }
+
+  /**
+   * Helper method to load all the users specified in the form state.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return \Drupal\user\UserInterface[]
+   *   An array of loaded user entities.
+   */
+  protected function getUserList(FormStateInterface $form_state): array {
+    $list = [];
+    foreach ($form_state->get('user_list') as $mail) {
+      $list[] = $this->loadUserByMail($mail);
+    }
+
+    return $list;
+  }
+
+  /**
+   * Loads a user by its mail.
+   *
+   * @param string $mail
+   *   The mail of the user.
+   *
+   * @return \Drupal\user\UserInterface|null
+   *   A loaded user object. Null if the mail matches no users in the system.
+   */
+  protected function loadUserByMail($mail): ?UserInterface {
+    $user = $this->entityTypeManager->getStorage('user')
+      ->loadByProperties(['mail' => $mail]);
+
+    return empty($user) ? NULL : reset($user);
+  }
+
   /**
    * Returns a full name of the user with his email as suffix in parenthesis.
    *
    * @param \Drupal\user\UserInterface $user
    *   The user object.
    *
-   * @return \Drupal\Core\StringTranslation\TranslatableMarkup
+   * @return string
    *   A string version of user's full name.
    */
-  protected function getAccountName(UserInterface $user) : TranslatableMarkup {
-    return $this->t('@name (@email)', [
-      '@name' => $user->get('full_name')->value,
-      '@email' => $user->getEmail(),
-    ]);
+  protected function getAccountName(UserInterface $user): string {
+    return $user->get('full_name')->value;
```

Comment:
I do not really like this drawback. I see that now when you start typing you get the full name and the user name in the parenthesis. I don't think most cases will be done using user names.
I would suggest that we do show the email even instead of the user name if we cannot show both. 
If there will be two people named "John Smith" with usernames "john_smith" and "smith_john" it can be very weird. And that's the case where the user names match and it is not starlord and monkey respectively. 
An e-mail would be absolute and it would also cover the case where the users communicated externally through the mailing system and the facilitator would have to distinguish the users.

Reply:
Please read description in the JIRA ticket for why the PM decided like this.